daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/processors/unparsers/SpecifiedLength2.scala

```diff
@@ -476,34 +476,48 @@ class ElementUnusedUnparser(
  * simply a non-negative integer.
  */
 class ChoiceUnusedUnparserSuspendableOperation(
-  override val rd: ChoiceRuntimeData,
-  targetLengthInBits: Long,
-  maybeDOSStartForChoiceUnused: Maybe[DataOutputStream],
-  maybeDOSEndForChoiceUnused: Maybe[DataOutputStream])
+  override val rd: ModelGroupRuntimeData,
+  targetLengthInBits: Long)
   extends SuspendableOperation
   with StreamSplitter
   with SkipTheBits {
 
   private var zlStatus_ : ZeroLengthStatus = ZeroLengthStatus.Unknown
```

Comment:
Here's the zero-length status use. This object is only created at runtime now, so yeah ZeroLengthStatus probably doesn't have to be serializable. That said. Doesn't hurt.